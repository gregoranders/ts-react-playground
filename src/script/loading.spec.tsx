import React, { act, create, mount } from '@app/testUtils';

import * as TestSubject from '@app/loading';

describe(`${TestSubject.Loading.displayName}`, () => {
  it('exports default', () => {
    expect(TestSubject.default).toBeDefined();
  });

  const TestComponent = TestSubject.default;

  it('snapshot', (): void => {
    const snapshot = create(<TestComponent />);
    expect(snapshot).toMatchSnapshot();
  });

  it('mount', () => {
    const stepForwardInTime = (steps = 10) => {
      act(() => {
        jest.advanceTimersToNextTimer(steps);
      });
    };
    const testSubject = mount(<TestComponent timeout={10} />);
    stepForwardInTime(10);
    expect(testSubject.find('h1').text()).toBe('·');
    stepForwardInTime(9);
    expect(testSubject.find('h1').text()).toBe('··');
    stepForwardInTime(9);
    expect(testSubject.find('h1').text()).toBe('···');
    stepForwardInTime(9);
    expect(testSubject.find('h1').text()).toBe('····');
    stepForwardInTime(9);
    expect(testSubject.find('h1').text()).toBe('·····');
    stepForwardInTime(9);
    expect(testSubject.find('h1').text()).toBe('····');
    stepForwardInTime(9);
    expect(testSubject.find('h1').text()).toBe('···');
    stepForwardInTime(9);
    expect(testSubject.find('h1').text()).toBe('··');
    stepForwardInTime(9);
    expect(testSubject.find('h1').text()).toBe('·');
    stepForwardInTime(9);
    expect(testSubject.find('h1').text()).toBe('··');
  });
});
