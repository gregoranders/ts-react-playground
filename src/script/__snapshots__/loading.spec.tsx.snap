// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading snapshot 1`] = `
<div
  className="centered bootstrap"
>
  <h1
    className="dots"
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</div>
`;
